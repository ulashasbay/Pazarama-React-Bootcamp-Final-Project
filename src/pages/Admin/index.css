.admin-login-container {
  font-family: "Open Sans", sans-serif;
  color: #3a3c47;
  line-height: 1.6;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0;
  padding: 0;
  height: auto;
}

.admin-login-container h1 {
  margin-top: 48px;
  font-size: 30px;
}

.login-form {
  background: #fff;
  max-width: 360px;
  width: 100%;
  padding: 58px 44px;
  border: 1px solid #e1e2f0;
  border-radius: 4px;
  box-shadow: 0 0 5px 0 rgba(42, 45, 48, 0.12);
  transition: all 0.3s ease;
}

.login-row {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.login-row label {
  font-size: 13px;
  color: #8086a9;
}

.login-row input {
  flex: 1;
  padding: 13px;
  border: 1px solid #d6d8e6;
  border-radius: 4px;
  font-size: 16px;
  transition: all 0.2s ease-out;
}

.login-row input:focus {
  outline: none;
  box-shadow: inset 2px 2px 5px 0 rgba(42, 45, 48, 0.12);
}

.login-row input::placeholder {
  color: #c8cddf;
}

/* text area alanları için */
.login-row textarea {
  flex: 1;
  padding: 13px;
  border: 1px solid #d6d8e6;
  border-radius: 4px;
  font-size: 16px;
  transition: all 0.2s ease-out;
}

.login-row textarea:focus {
  outline: none;
  box-shadow: inset 2px 2px 5px 0 rgba(42, 45, 48, 0.12);
}

.login-row textarea::placeholder {
  color: #c8cddf;
}

.login-errors {
  margin-top: 0;
  font-size: 14px;
  color: red;
}

@media (max-width: 960px) {
  .admin-login-container {
    margin: 0 18px;
  }

  .login-form {
    border: none;
    box-shadow: none;
    padding: 20px 0;
  }
}
