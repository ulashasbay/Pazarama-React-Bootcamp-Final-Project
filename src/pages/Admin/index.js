import React from "react";
import { useFormik } from "formik";
import Input from "../../components/Input";
import FormSubmitBtn from "../../components/FormSubmitBtn";
import AdminLoginVal from "../../schema/AdminLoginVal";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { updateIsLoggedIn } from "../../redux/isLoggedIn/isLoggedInSlice";
import "./index.css";

function Admin() {
  let navigate = useNavigate();
  const dispatch = useDispatch();

  const { handleSubmit, handleChange, handleBlur, errors, touched } = useFormik(
    {
      initialValues: {
        username: "",
        password: "",
      },
      onSubmit: (values) => {
        if (
          values.username === "kodluyoruz" &&
          values.password === "bootcamp109"
        ) {
          localStorage.setItem("username", "kodluyoruz");
          localStorage.setItem("password", "bootcamp109");
          dispatch(updateIsLoggedIn(true));
          navigate("/admin/basvuru-listesi");
        }
      },
      validationSchema: AdminLoginVal,
      validateOnChange: false,
      validateOnBlur: false,

    }
  );

  return (
    <>
      <div className="admin-login-container">
        <h1>Giriş Yap</h1>
        <form className="login-form" onSubmit={handleSubmit}>
          <div className="login-row">
            <Input
              name="username"
              type="text"
              text="Kullanıcı Adı: kodluyoruz"
              placeholder="Kullanıcı Adı"
              onChange={handleChange}
              onBlur={handleBlur}
            />
            {errors.username && touched.username && (
              <span className="login-errors">{errors.username}</span>
            )}
          </div>
          <div className="login-row">
            <Input
              name="password"
              type="password"
              text="Şifre: bootcamp109"
              placeholder="Şifre"
              onChange={handleChange}
              onBlur={handleBlur}
            />
            {errors.password && touched.password && (
              <span className="login-errors">{errors.password}</span>
            )}
          </div>
          <FormSubmitBtn text="Giriş" />
        </form>
      </div>
    </>
  );
}

export default Admin;
