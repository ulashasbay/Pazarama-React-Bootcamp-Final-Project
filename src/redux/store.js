import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
