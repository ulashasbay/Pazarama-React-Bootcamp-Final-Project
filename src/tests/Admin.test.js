import React from "react";
import { render, screen } from "@testing-library/react";
import Admin from "../pages/Admin";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("Admin Tests", () => {
  let usernameInput, passwordInput, button;
  beforeEach(() => {
    render(<Admin />, { wrapper: AllTheProviders });
    usernameInput = screen.getByPlaceholderText("Kullanıcı Adı");
    passwordInput = screen.getByPlaceholderText("Şifre");
    button = screen.getByText("Giriş");
  });

  test("Render Header", () => {
    const header = screen.getByText("Giriş Yap");
    expect(header).toBeInTheDocument();
  });

  test("Render form labels", () => {
    const usernameLabel = screen.getByText("Kullanıcı Adı");
    expect(usernameLabel).toBeInTheDocument();

    const passwordLabel = screen.getByText("Şifre");
    expect(passwordLabel).toBeInTheDocument();
  });

  test("Render Inputs", () => {
    expect(usernameInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
  });

  test("Kullanıcı Adı input type is text", () => {
    expect(usernameInput.type).toEqual("text");
  });

  test("password input type is password", () => {
    expect(passwordInput.type).toEqual("password");
  });

  test("Render Giriş Button", () => {
    expect(button).toBeInTheDocument();
  });

  test("Giriş button type is submit", () => {
    expect(button.type).toEqual("submit");
  });
});

test("Admin renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Admin />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
