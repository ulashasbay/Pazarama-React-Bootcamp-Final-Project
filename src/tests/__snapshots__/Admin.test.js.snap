// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin renders correctly 1`] = `
<div
  className="admin-login-container"
>
  <h1>
    Giriş Yap
  </h1>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="login-row"
    >
      <label
        htmlFor="username"
      >
        Kullanıcı Adı
      </label>
      <input
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Kullanıcı Adı"
        type="text"
      />
    </div>
    <div
      className="login-row"
    >
      <label
        htmlFor="password"
      >
        Şifre
      </label>
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Şifre"
        type="password"
      />
    </div>
    <button
      className="app-check-button"
      type="submit"
    >
      Giriş
    </button>
  </form>
</div>
`;
